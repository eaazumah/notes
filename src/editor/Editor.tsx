import React from 'react';
import styled from 'styled-components/native';
import Container from '../@shared/styled/Container';
import ScrollView from '../@shared/styled/ScrollView';
import Parser from '../parser/Parser';
import Header from './components/Header';
import useEditorLogic from './hooks/useEditorLogic';

interface Props {}

const Editor: React.FC<Props> = () => {
  const logic = useEditorLogic();
  return (
    <Container>
      <Header
        isArchived={logic.isArchived}
        isFavorite={logic.isFavorite}
        toggleNoteArchived={logic.toggleNoteArchived}
        toggleNoteFavorite={logic.toggleNoteFavorite}
      />
      <ScrollView showsVerticalScrollIndicator={false}>
        <Title
          multiline
          value={logic.note?.title}
          placeholder={'Title'}
          onChangeText={logic.updateNoteTitle}
        />
        <Content
          multiline
          autoFocus
          placeholder={'Note'}
          onChangeText={logic.handlerTextChange}
          onSelectionChange={logic.onSelectionChange}>
          <Parser opacity={1}>{logic.note?.text}</Parser>
        </Content>
      </ScrollView>
    </Container>
  );
};

const Title = styled.TextInput.attrs(props => ({
  placeholderTextColor: props.theme.primaryText,
}))`
  font-size: 18px;
  margin-top: 5px;
  font-weight: bold;
  margin-bottom: 10px;
  color: ${props => props.theme.primaryText};
`;

const Content = styled.TextInput.attrs(props => ({
  scrollEnabled: false,
  placeholderTextColor: props.theme.primaryText,
}))``;

export default Editor;
