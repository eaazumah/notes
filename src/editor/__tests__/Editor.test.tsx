// import { useRoute } from '@react-navigation/core';
import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import Editor from '../Editor';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useRoute: () => ({
      key: 'string',
      name: 'string',
      params: {noteId: 'string'},
    }),
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

describe('Editor', () => {
  test('Editor renders correctly', () => {
    const screen = render(
      <TestProvider>
        <Editor />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('Editor functionality test', async () => {
    const screen = render(
      <TestProvider>
        <Editor />
      </TestProvider>,
    );
    expect(screen.queryByTestId('note-input')).toBeNull();

    const toggleEdit = screen.queryByTestId('toggle-edit');
    expect(toggleEdit).toBeTruthy();
    fireEvent.press(toggleEdit!);

    expect(screen.queryByTestId('note-input')).toBeTruthy();
  });
});
