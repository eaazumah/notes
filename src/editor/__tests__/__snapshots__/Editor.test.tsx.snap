// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Editor renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 50,
          "paddingHorizontal": 17,
          "width": "100%",
        },
      ]
    }
    testID="header-editor"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Icon
        color="#242A37"
        name="arrow-back"
        size={25}
        testID="icon"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Icon
          color="#242A37"
          name="bookmark"
          size={25}
          testID="icon"
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Icon
          color="#242A37"
          name="archive-arrow-down-outline"
          size={25}
          testID="icon"
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {
          "backgroundColor": "#F5F5F5",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingHorizontal": 18,
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <HTMLView
        rootComponentProps={
          Object {
            "maxHeight": undefined,
            "overflow": "hidden",
          }
        }
        style={
          Array [
            Object {},
          ]
        }
        textComponentProps={
          Object {
            "style": Object {
              "color": "#242A37",
            },
          }
        }
        value=""
      />
    </View>
  </RCTScrollView>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "paddingBottom": 17,
            "paddingLeft": 17,
            "paddingRight": 17,
            "paddingTop": 17,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="toggle-edit"
      >
        <Icon
          color="#242A37"
          name="edit"
          size={25}
          testID="icon"
        />
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
